app/modules/projects/parameters.py

```diff
@@ -16,8 +15,32 @@ class Meta(schemas.DetailedProjectSchema.Meta):
         pass
 
 
-class PatchProjectDetailsParameters(PatchJSONParameters):
+class PatchProjectDetailsParameters(PatchJSONParametersWithPassword):
     # pylint: disable=abstract-method,missing-docstring
-    OPERATION_CHOICES = (PatchJSONParameters.OP_REPLACE,)
 
-    PATH_CHOICES = tuple('/%s' % field for field in (Project.title.key,))
+    # Valid options for patching are '/title', '/UserAdd', '/UserRemove', '/EncounterAdd' and '/EncounterRemove'.
+    # The '/current_password' is not patchable but must be a valid field in the patch so that it can be
+    # present for validation
+
+    VALID_FIELDS = [
+        Project.title.key,
+        'current_password',
+        'UserAdd',
+        'UserRemove',
+        'EncounterAdd',
+        'EncounterRemove',
+    ]
```

Comment:
I was also experimenting with something like the VALID_FIELDS you have here, an waffling a little on how to define them. Since we can get the literal field name (users or encounters) and the op, maybe we can pass that in and keep our calls like: {"op":"add", "path":"/encounters", "value":"[UUID'S]"} instead of {"op":"add", "path":"/EncounterAdd", "value":"[UUID'S]"}.

Reply:
I would make a strong vote for keeping the op looking like this:
```
operations = [
{"op":"add", "path":"/encounter", "value":"<ENCOUNTER GUID 1>"},
{"op":"add", "path":"/encounter", "value":"<ENCOUNTER GUID 2>"},
{"op":"add", "path":"/encounter", "value":"<ENCOUNTER GUID 3>"},
{"op":"remove", "path":"/user_by_email", "value":"<USER EMAIL 1>"},
{"op":"add", "path":"/user", "value":"<USER GUID 1>"},
]
```

Reply:
@bluemellophone I'd have liked that too, see thread on dev-houston, the problem is that remove only has a field, it doesn't have a value.
github.com/WildMeOrg/houston/blob/master/flask_restplus_patched/parameters.py#L170

I was avoiding updating the flask_restplus_patched as I thought that was inherited from elsewhere.
But I think that is now justified to update the remove interface to have a value, I will do that instead.

Reply:
the rfc for patch definitely _does_ seem to simply omit the **value** part for **_remove_**.   !!
there is a [healthy discussion about this here](https://stackoverflow.com/questions/27439986/what-is-the-json-patch-format-to-remove-an-element-from-an-array), which basically says one is "supposed to" effectively append what we are talking about as value as a suffix to the path.  in the example, there are using the _index_ (int) of the value within the array!   in other words, one would do something like:  `"path": "/user/5"`   :grimacing:  this isnt really cool for unordered sets and other issues that the above thread gets into, so.....

if we truly wanted to stick to the rfc, maybe we could kinda go with this _idea_ though, and do something like: `"path": "/user_by_email/my@email"` and then `"path": "/user/GUID"` alternately.... ???

Reply:
Ah, I see that this rabbit hole is quite deep and might contain a few wolves. 
So we either piggy back a value onto the end of the path or do it the non standard way I did so. 
If this is between our frontend and our houston, do we have to see the RFC as definitive or as a guideline.

Reply:
I'd argue guideline.  This PR allows a value to be specified to the `remove` patch operation, in conflict with the RFC.  It's a good thing we are the developers and it better sits our needs.  Let's add an explicit notice to the README and the documentation for the API that we conform to the RFC except for this one case

Reply:
yeah good call on the addressing things if/when a **need** arises (e.g. client pops up).